tests/lambda_functions/nightly_stats_bigquery_upload/test_nightly_lambda.py

```diff
@@ -0,0 +1,319 @@
+    )
+
+    mock_client = mocker.Mock(Client)
+    mock_bigquery.Client.return_value = mock_client
+
+    return mock_client
+
+
+@pytest.fixture(autouse=True)
+def mock_credentials(mocker):
+    mock_service_account = mocker.patch(
+        'lambda_functions.nightly_stats_bigquery_upload.nightly_stats_bigquery_upload_lambda.service_account'
+    )
+
+    mock_service_account.Credentials.from_service_account_info.return_value = mocker.Mock(Credentials)
+
+
+class TestParameterAccess:
+    @staticmethod
+    def test_read_service_account_info_from_ssm(mock_ssm_client) -> None:
+        assert nightly_lambda.read_service_account_info_from_ssm() == EXAMPLE_SERVICE_ACCOUNT_INFO
+
+
+class TestEventParsing:
+    @staticmethod
+    @pytest.mark.parametrize(
+        'example_event, expected_object_key',
+        [
+            (EXAMPLE_S3_EVENT_STATS, OBJECT_KEY_STATS),
+            (EXAMPLE_S3_EVENT_BILLING, OBJECT_KEY_BILLING),
+        ],
+        ids=['stats', 'billing'],
+    )
+    def test_get_object_key(example_event, expected_object_key) -> None:
+        assert nightly_lambda.get_object_key(example_event) == expected_object_key
+
+    @staticmethod
+    def test_get_bucket_name() -> None:
+        assert nightly_lambda.get_bucket_name(EXAMPLE_S3_EVENT_STATS) == BUCKET_NAME
+
+
+class TestS3Access:
+    @staticmethod
+    @pytest.mark.parametrize(
+        'object_key, expected_response',
+        [
+            (OBJECT_KEY_STATS, example_nightly_stats_bytes()),
+            (OBJECT_KEY_BILLING, example_nightly_billing_bytes()),
+        ],
+        ids=['stats', 'billing'],
+    )
+    def test_read_nightly_stats_from_s3(mock_s3_client, object_key, expected_response) -> None:
+        response = nightly_lambda.read_nightly_stats_from_s3(BUCKET_NAME, object_key)
+        assert response == expected_response
+
+
+class TestBigQueryAccess:
+    @staticmethod
+    def test_delete_existing_rows_for_date(mock_bigquery_client) -> None:
+        date, _type, _ext = OBJECT_KEY_STATS.split('.')
+
+        nightly_lambda.delete_existing_rows_for_date(mock_bigquery_client, BQ_TABLE_ID, date)
+
+        mock_bigquery_client.query.assert_called_once_with(f"DELETE FROM `{BQ_TABLE_ID}` WHERE date = '{date}'")
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        'bq_table_id, example_nightly_bytes',
+        [
+            (nightly_lambda.TABLE_ID_STATS, example_nightly_stats_bytes()),
+            (nightly_lambda.TABLE_ID_BILLING, example_nightly_billing_bytes()),
+        ],
+        ids=['stats', 'billing'],
+    )
+    def test_add_updated_rows_for_date(mock_bigquery_client, bq_table_id, example_nightly_bytes) -> None:
+        nightly_lambda.add_updated_rows_for_date(mock_bigquery_client, bq_table_id, example_nightly_bytes)
+
+        _, kwargs = mock_bigquery_client.load_table_from_file.call_args
+
+        assert kwargs['destination'] == bq_table_id
+        assert kwargs['file_obj'].getvalue() == example_nightly_bytes
```

Comment:
I suggest naming the variable something other than "kwargs," which is usually associated with a function signature.

Reply:
It's coming from `.call_args` so it seems appropriate to me. If you have a name you'd like to see let me know, otherwise I'm inclined to leave it.